tools/refguide_check.py

```diff
@@ -755,7 +755,10 @@ def check_doctests_testfile(fname, verbose, ns=None,
         return results
 
     full_name = fname
-    text = open(fname).read()
+    if sys.version_info.major <= 2:
+        text = open(fname).read()
+    else:
+        text = open(fname, encoding='utf-8').read()
```

Comment:
This could probably be:
```
with codecs.open(fname, encoding='utf-8') as fid:
    text = fid.read()
```
That should work on 2 and 3, and properly close the file.

Reply:
Would that not change the semantics slightly?  I believe that codecs.open() opens in binary mode and doesn't do '\n' conversion, and in Py2 it would always return Unicode.  I did think about using `io.open(fname, encoding='utf-8')` but didn't because `text` would now be a Unicode object in Py2 and downstream code may not be expecting Unicode.

Reply:
Well if it broke things we should probably fix those downstream things to handle unicode. But for now (esp. since Py2 support will be dropped within a year) we can go with your solution, but please change it to use the context manager to properly close the file.